src/atomate2/openmm/jobs/base.py

```diff
@@ -0,0 +1,156 @@
+from typing import Optional, Union, Callable
+from jobflow import job, Maker
+from dataclasses import dataclass
+from pathlib import Path
+from pymatgen.io.openmm.sets import OpenMMSet
+from src.atomate2.openmm.schemas.openmm_task_document import OpenMMTaskDocument
+from openmm import (
+    Platform,
+    Context,
+)
+from typing import (
+    Union,
+    Optional,
+    Dict,
+)
+from openmm.app import (
+    DCDReporter,
+    StateDataReporter,
+    PDBReporter,
+)
+from openmm.openmm import (
+    MonteCarloBarostat,
+    AndersenThermostat,
+)
+from openmm.unit import (
+    kelvin,
+    atmosphere,
+    pico,
+    second
+)
+import os
+
+
+def openmm_job(method: Callable):
+    """
+
+    Parameters
+    ----------
+    method : callable
+        A BaseOpenmmMaker.make method. This should not be specified directly and is
+        implied by the decorator.
+
+    Returns
+    -------
+    callable
+        A decorated version of the make function that will generate OpenMM jobs.
+    """
+    return job(method, output_schema=OpenMMTaskDocument)
+
+
+@dataclass
+class BaseOpenmmMaker(Maker):
+    """
+    todo: doc string
+    """
+    name: str = "base openmm job"
+    get_simulation_kwargs: Optional[dict] = None
+    dcd_reporter_kwargs: Optional[dict] = None
+    pdb_reporter_kwargs: Optional[dict] = None
+    state_reporter_kwargs: Optional[dict] = None
+    pressure_barostat_kwargs: Optional[dict] = None
+    temperature_barostat_kwargs: Optional[dict] = None
```

Comment:
I think this makes sense for reporters but not for the state, pressure, and temp stuff. Those should be tailored to the individual jobs.

Reply:
Upon more reflection here, let's not provide the option to use the PDB reporter. It's a very inefficient way to store the trajectory and it doesn't offer any more information than the DCD.

Reply:
Roger. I'll address in the next commit